lib/Word_Lib/Word_Lemmas.thy

```diff
@@ -1466,8 +1466,11 @@ lemma mask_shift_sum:
   "\<lbrakk> a \<ge> b; unat n = unat (p AND mask b) \<rbrakk>
    \<Longrightarrow> (p AND NOT(mask a)) + (p AND mask a >> b) * (1 << b) + n = (p :: 'a :: len word)"
   apply (simp add: shiftl_def shiftr_def flip: push_bit_eq_mult take_bit_eq_mask word_unat_eq_iff)
-  apply (subst disjunctive_add, clarsimp simp add: bit_simps)+
-  apply (rule bit_word_eqI)
+  apply (subst disjunctive_add)
+   apply (auto simp add: bit_simps)
```

Comment:
Limit the auto to the current subgoal? And below

Reply:
I have the felling that'll break the proof. So great to have stuff in the AFP..

Reply:
Ok, managed to fix that one and a few more of them.